tests/cupy_tests/test_ndarray_elementwise_op.py

```diff
@@ -264,3 +265,42 @@ def test_array_reversed_sub(self):
 
     def test_array_reversed_mul(self):
         self.check_array_reversed_op(operator.mul)
+
+    @testing.for_all_dtypes(no_bool=True)
+    def check_typecast(self, val, dtype):
+        a = val + testing.shaped_arange((5,), numpy, dtype)
+        b = val + testing.shaped_arange((5,), cupy, dtype)
```

Comment:
I think `(shaped_arange((5,)) - 2)` will make stronger test case than `shaped_arange((5,))` because the former checks more boundary cases.